File: Resueltos_por_competidor/Tomás_Triana_Galvis/CountingDivisors.cpp
#include<bits/stdc++.h>
typedef long long ll;
using namespace std;

/*
 * Autor: Tomás Triana Galvis
 * Problema: Counting Divisors
 * Juez online: CSES1713
 * Veredicto: Accepted
 * Url: https://cses.fi/problemset/task/1713/
 **/ 
int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    int t,n;
    vector<int> ans(1000000);
    for(int i = 1;i<=1000000;i++){
        for(int j = i; j<=1000000;j+=i){
            ans[j]++;
        }
    }
    cin>>t;
    while(t--){
        cin>>n;
        cout<<ans[n]<<endl;
    }
    return 0;
}

File: Resueltos_por_competidor/Juan_Morales/CountingDivisors.java
/*
 * Autor: OtherSpanish
 * Problema: Counting Divisors
 * Juez online: CSES
 * Veredicto: Accepted
 * URL: https://cses.fi/problemset/task/1713
 */

import java.util.Scanner;

public class Main
{
	public static void main(String[] args) {
    
    int MAX = 1000001;
    
    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();
    int div[] = new int[MAX];
    for (int d = 1; d < MAX; d++) {
		for (int i = d; i < MAX; i+=d) {
			div[i]++;
		}
	}
    StringBuilder sb = new StringBuilder();
    for (int i = 0; i < n; i++) {
		int x = sc.nextInt();
		sb.append(div[x]).append("\n");
	}
    System.out.println(sb);
	}
}
